homeassistant/components/samsungtv/__init__.py

```diff
@@ -301,9 +301,15 @@ async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
             for device in dr.async_entries_for_config_entry(
                 dev_reg, config_entry.entry_id
             ):
-                for connection in device.connections:
-                    if connection == (dr.CONNECTION_NETWORK_MAC, "none"):
-                        dev_reg.async_remove_device(device.id)
+                new_connections = {
+                    c
+                    for c in device.connections
+                    if c != (dr.CONNECTION_NETWORK_MAC, "none")
+                }
```

Comment:
It would be slightly faster to copy the set and discard the specific key

Reply:
I don't have access to a PC until Monday now :(
A suggestion would be appreciated